### PR TITLE
feat(rust): Alternative hybrid consumer that uses RunTaskWithMultiprocessing

### PR DESCRIPTION
**Pull request by lynnagara**
The current hybrid consumer that uses procspawn is deadlocking. This is an alternative approach to hybrid which uses the existing RunTaskWithMultiprocessing strategy from Python Arroyo from a Rust consumer.

All hybrid consumer code is considered highly experimental.

### CODE REVIEW

File: rust_snuba/src/strategies/python_v2.rs [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    }
+
+    fn submit(&mut self, message: Message<KafkaPayload>) -> Result<(), SubmitError<KafkaPayload>> {
+        if self.transformed_messages.len() > 100_000 {
+            return Err(SubmitError::MessageRejected(MessageRejected { message }));
+        }
+
+        let mut apply_backpressure = false;
+
+        let _ = Python::with_gil(|py| -> PyResult<()> {
```
Comment (untitaker) [inline rust_snuba/src/strategies/python_v2.rs:143]: please handle this result. you can probably just unwrap here

File: snuba/consumers/rust_processor.py [hunk 2]
```diff
@@ -66,3 +77,171 @@ def process_rust_message(
...
+        """
+        self.__inner.close()
+        try:
+            self.__inner.join(timeout=timeout)
+        except InvalidMessage:
+            # We cannot forward offsets here since the inner strategy is already
+            # marked as closed. Log the exception and move on. The message will get
+            # reprocessed properly when the consumer is restarted.
+            logger.warning("Invalid message in join", exc_info=True)
+        self.__pool.close()
```
Comment (untitaker) [inline snuba/consumers/rust_processor.py:242]: i think you need to close or terminate the pool in `terminate` too

**Review by untitaker — CHANGES_REQUESTED** [review]
let's just deploy this for now since it's not running on any prod data. some comments inline but otherwise lgtm

File: rust_snuba/src/strategies/python_v2.rs (hunk 1 above)
Comment (lynnagara) [reply rust_snuba/src/strategies/python_v2.rs]: Updated

**Review by untitaker — APPROVED** [review]
approving so you can merge async but the above review comments are imo important

**Pull request merged**